src/drt/src/dr/FlexDR.cpp

```diff
@@ -696,121 +1241,61 @@ void FlexDR::searchRepair(const SearchRepairArgs& args)
-                j = (j + 1) % router_->getCloudSize();
-              }
-            }
-            {
-              ProfileTask task("DIST: SERIALIZE+SEND");
-#pragma omp parallel for schedule(dynamic)
-              for (int i = 0; i < distWorkerBatches.size(); i++)  // NOLINT
-                sendWorkers(distWorkerBatches.at(i), workersInBatch);
-            }
-            logger_->report("    Received Batches:{}.", t);
-            std::vector<std::pair<int, std::string>> workers;
-            router_->getWorkerResults(workers);
-            {
-              ProfileTask task("DIST: DESERIALIZING_BATCH");
-#pragma omp parallel for schedule(dynamic)
-              for (int i = 0; i < workers.size(); i++) {  // NOLINT
-                deserializeWorker(workersInBatch.at(workers.at(i).first).get(),
-                                  design_,
-                                  workers.at(i).second);
-              }
-            }
-            logger_->report("    Deserialized Batches:{}.", t);
-          }
-        }
-      }
-      {
-        ProfileTask profile("DR:end_batch");
-        // single thread
-        for (auto& worker : workersInBatch) {
-          if (worker->end(getDesign())) {
-            numWorkUnits_ += 1;
-          }
-          if (worker->isCongested()) {
-            increaseClipsize_ = true;
-          }
+          processWorkersBatchDistributed(workersInBatch, version, iter_prog);
+        } else {
+          processWorkersBatch(workersInBatch, iter_prog);
         }
-        workersInBatch.clear();
       }
+      endWorkersBatch(workersInBatch);
     }
   }
 
-  if (!iter) {
+  if (!iter_) {
     removeGCell2BoundaryPin();
   }
-  if (router_cfg_->VERBOSE > 0) {
-    if (cnt * 1.0 / tot >= prev_perc / 100.0 + 0.1 && prev_perc >= 90) {
-      if (prev_perc == 0 && t.isExceed(0)) {
-        isExceed = true;
-      }
-      prev_perc += 10;
-      if (isExceed) {
-        logger_->report("    Completing {}% with {} violations.",
-                        prev_perc,
-                        getDesign()->getTopBlock()->getNumMarkers());
-        logger_->report("    {}.", t);
-      }
+}
+
+void FlexDR::searchRepair(const SearchRepairArgs& args)
+{
+  const RipUpMode ripupMode = args.ripupMode;
+  if ((ripupMode == RipUpMode::DRC || ripupMode == RipUpMode::NEARDRC)
+      && getDesign()->getTopBlock()->getMarkers().empty()) {
+    return;
+  }
+  ProfileTask profile(fmt::format("DR:searchRepair{}", iter_).c_str());
+
+  if (dist_on_) {
+    if ((iter_ % 10 == 0 && iter_ != 60) || iter_ == 3 || iter_ == 15) {
+      router_cfg_path_ = fmt::format("{}globals.{}.ar", dist_dir_, iter_);
+      router_->writeGlobals(router_cfg_path_);
     }
   }
+  // start timer for the current iteration
+  IterationProgress iter_prog;
+  auto block = getDesign()->getTopBlock();
+  const auto num_drvs = block->getNumMarkers();
+  const bool stubborn_flow = num_drvs <= 11 && ripupMode != RipUpMode::ALL
```

Comment:
why 11? would this be better as a function of DRCs per unit area (it seems like 11 would be a large number for small design and potentially a small number for a large design)?

Reply:
This is an initial attempt at this flow. The criteria for deciding when to start this flow is still a work in progress. However, as an initial attempt, this seems to be beneficial for now.
In my opinion the number of DRCs should not be the deciding factor here. Let's say we have 1000 DRCs. if we look at the number of DRCs only, we would not attempt this flow. Now, let's assume these 1000 DRCs are at the same Gcell. Now it would be fit to start this flow.
I am still working on this part, but I think it's ok to start with this PR.